project/src/main/java/com/google/sps/servlets/Announcements.java

```diff
@@ -46,52 +47,46 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-      String name = "";
-      String school = "";
-      Query query = new Query("User");
-      PreparedQuery results = datastore.prepare(query);
-      for(Entity entity:results.asIterable()){
-          if (((String) entity.getProperty("email")).toLowerCase().equals(userService.getCurrentUser().getEmail().toLowerCase())) {
-              name = (String) entity.getProperty("name");
-              school = (String) entity.getProperty("school");
-          }
-      }
+      HttpSession session = request.getSession(false); 
+      String n = (String) session.getAttribute("person");
+      String admin = n.substring(n.indexOf("admin\":")+7, n.indexOf("}"));
+      String name = n.substring(n.indexOf("name\":")+7, n.indexOf("\",\"userId"));
+      String school = n.substring(n.indexOf("school\":")+9, n.indexOf("\",\"phone"));
       String title = request.getParameter("title");
       String content = request.getParameter("content");
+      String importance = request.getParameter("importance");
       Entity commentEntity = new Entity("Announcements");
       java.util.Date now=new java.util.Date();
       commentEntity.setProperty("from", name);
       commentEntity.setProperty("title", title);
       commentEntity.setProperty("content", content);
       commentEntity.setProperty("when", now);
       commentEntity.setProperty("school", school);
-      datastore.put(commentEntity);
+      commentEntity.setProperty("importance", importance);
+      if (admin.equals("true")) {
```

Comment:
does it make more sense to name after`is_admin`?

Reply:
I agree, done!